File: 30 Days of Code/Day 17 More Exceptions/Day 17 More Exceptions.cpp
#include <cmath>
#include <iostream>
#include <exception>
#include <stdexcept>
using namespace std;

class Calculator
{
    public: int power(int n, int p) 
    {
        if (n < 0 || p < 0)
            throw std::runtime_error("n and p should be non-negative");
        else
        {
            int res = 1;
            for (int i = 0; i < p; i++)
            {
                res *= n;
            }
            return res;
        }
    }
};

int main()
{
    Calculator myCalculator=Calculator();
    int T,n,p;
    cin>>T;
    while(T-->0){
      if(scanf("%d %d",&n,&p)==2){
         try{
               int ans=myCalculator.power(n,p);
               cout<<ans<<endl; 
         }
         catch(exception& e){
             cout<<e.what()<<endl;
         }
      }
    }
    
}

File: 30 Days of Code/Day 17 More Exceptions/Day 17 More Exceptions.cs
using System;

public class Calculator
{
    public int power(int n, int p) 
    {
        if (n < 0 || p < 0)
            throw new Exception("n and p should be non-negative");
        else
        {
            int res = 1;
            for (int i = 0; i < p; i++)
            {
                res *= n;
            }
            return res;
        }
    }
}

class Solution{
    static void Main(String[] args){
        Calculator myCalculator=new  Calculator();
        int T=Int32.Parse(Console.ReadLine());
        while(T-->0){
            string[] num = Console.ReadLine().Split();
            int n = int.Parse(num[0]);
            int p = int.Parse(num[1]); 
            try{
                int ans=myCalculator.power(n,p);
                Console.WriteLine(ans);
            }
            catch(Exception e){
               Console.WriteLine(e.Message);

            }
        }
        
        
        
    }
}